File: CS/1389_Create_Target_Array_in_the_Given_Order.cs
// 2022-03-06 12:34~12:37

public class Solution {
    public int[] CreateTargetArray(int[] nums, int[] index) {
        IList<int> result = new List<int>();
        for(int i = 0; i < nums.Length; i++){
            result.Insert(index[i], nums[i]);
        }
        return result.ToArray();
    }
}

File: Cpp/1389_Create_Target_Array_in_the_Given_Order.cpp
class Solution {
public:
    vector<int> createTargetArray(vector<int>& nums, vector<int>& index) {
        vector<int> result(nums.size());
        for(int i = 0; i < nums.size(); i++){
            for(int j = i; j > index[i]; j--){
                result[j] = result[j-1];
            }
            result[index[i]] = nums[i];
        }
        return result;
    }
};
